src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -363,8 +368,8 @@ struct areatrigger_pri_divine_star : AreaTriggerAI
         {
             _casterCurrentPosition = caster->GetPosition();
 
-            // Note: max. distance at which the Divine Star can travel to is 24 yards.
-            float divineStarXOffSet = 24.0f;
+            // Note: max. distance at which the Divine Star can travel to is EFFECT_1's BasePoints yards.
+            float divineStarXOffSet = sSpellMgr->GetSpellInfo(at->GetSpellId(), DIFFICULTY_NONE)->GetEffect(EFFECT_1).CalcValue(at->GetCaster());
```

Comment:
is there any validation needed before or after calling GetEffect(EFFECT_1) ?

Reply:
I know I can check whether the spell exists in db2 or not. The issue is I'm not sure how to validate a spell's effect in an areatrigger spell since we normally validate spells in SpellScripts.

`        SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(at->GetSpellId(), DIFFICULTY_NONE);
        if (!spellInfo)
            return;`

Reply:
I guess that works to validate, check it!